cohort-engine/src/main/java/com/ibm/cohort/engine/measure/seed/MeasureEvaluationSeeder.java

```diff
@@ -0,0 +1,175 @@
+	}
+
+	public MeasureEvaluationSeeder disableDebugLogging() {
+		this.debugMode = false;
+
+		return this;
+	}
+
+	public MeasureEvaluationSeeder disableExpressionCaching() {
+		this.disableExpressionCaching = true;
+
+		return this;
+	}
+
+
+	public MeasureEvaluationSeeder withTerminologyProvider(String source, String user, String password) {
+		this.terminologyProviderSource = source;
+		this.terminologyProviderUser = user;
+		this.terminologyProviderPassword = password;
+
+		return this;
+	}
+
+
+	public IMeasureEvaluationSeed create(Measure measure, String periodStart, String periodEnd, String productLine) {
+		LibraryHelper.loadLibraries(measure, this.libraryLoader, this.libraryResourceProvider);
+
+		// resolve primary library
+		org.cqframework.cql.elm.execution.Library library = LibraryHelper.resolvePrimaryLibrary(measure, libraryLoader, this.libraryResourceProvider);
+
+		List<Triple<String, String, String>> usingDefs = UsingHelper.getUsingUrlAndVersion(library.getUsings());
+
+		if (usingDefs.size() > 1) {
+			throw new IllegalArgumentException(
+					"Evaluation of Measure using multiple Models is not supported at this time.");
+		}
+
+
+		TerminologyProvider terminologyProvider = createTerminologyProvider(usingDefs);
+		LinkedHashMap<Triple<String, String, String>, DataProvider> dataProviders = createDataProviders(usingDefs, terminologyProvider);
+		Interval measurementPeriod = createMeasurePeriod(periodStart, periodEnd);
+		Context context = createContext(library, dataProviders, terminologyProvider, measurementPeriod, productLine);
+
+		// todo: [daniel.kim] this only returns the last seen data provider
+		// is this a bug?
+		List<Map.Entry<Triple<String, String, String>, DataProvider>> dataProviderList = new ArrayList<>(dataProviders.entrySet());
+		DataProvider lastDataProvider = dataProviderList.get(dataProviderList.size() - 1).getValue();
+
+		return new CustomMeasureEvaluationSeed(measure, context, measurementPeriod, lastDataProvider);
+	}
+
+	private Context createContext(
+			org.cqframework.cql.elm.execution.Library library,
+			Map<Triple<String, String, String>, DataProvider> dataProviders,
+			TerminologyProvider terminologyProvider,
+			Interval measurementPeriod,
+			String productLine) {
+		// todo: [daniel.kim] make sure all context.set calls are accounted for
+		Context context = new Context(library);
+		context.registerLibraryLoader(libraryLoader);
+
+		if (!dataProviders.isEmpty()) {
+			context.registerTerminologyProvider(terminologyProvider);
+		}
+
+		for (Map.Entry<Triple<String, String, String>, DataProvider> dataProviderEntry : dataProviders.entrySet()) {
+			Triple<String, String, String> usingDef = dataProviderEntry.getKey();
+			DataProvider dataProvider = dataProviderEntry.getValue();
+
+			context.registerDataProvider(usingDef.getRight(), dataProvider);
+		}
+
+		// todo: [daniel.kim] how is this different from the MeasurePeriod field?
+		context.setParameter(null, "Measurement Period",
```

Comment:
I would like the name of this to default to "Measurement Period", but be configurable by the client. I'm going to push to keep the "standard" name, but I'm not sure we can 100% rely on it.

Reply:
I agree that the recreation of the interval object is strange. The only thing I can think of is that they wanted the context to have its own private copy of the variable so that if the engine messed with it, it wouldn't get change the original value in MeasurementSeed.